test/test_tensorclass_nofuture.py

```diff
@@ -533,20 +729,29 @@ class MyDataParent:
     X_clone = X.clone()
     td = TensorDict({}, batch_size)
     td_clone = td.clone()
-    data_nest = MyDataNest(X=X, batch_size=batch_size)
-    data = MyDataParent(X=X, y=data_nest, z=td, W=W, batch_size=batch_size)
+    data_nest = MyDataNest(X=X, v="test_nested", batch_size=batch_size)
+    v = "test_tensorclass"
+    data = MyDataParent(X=X, y=data_nest, z=td, W=W, v=v, batch_size=batch_size)
     data_nest_clone = data_nest.clone()
     assert type(data_nest_clone) is type(data_nest)
     data.y = data_nest_clone
-    assert data.tensordict["y"] is not data_nest.tensordict
-    assert data.tensordict["y"] is data_nest_clone.tensordict, (
-        type(data.tensordict["y"]),
+    assert data.y.tensordict is not data_nest.tensordict
+    assert data.y.tensordict is data_nest_clone.tensordict, (
+        type(data.y.tensordict),
         type(data_nest.tensordict),
     )
     data.X = X_clone
     assert data.tensordict["X"] is X_clone
     data.z = td_clone
     assert data.tensordict["z"] is td_clone
+    v_new = "test_bluff"
+    data.v = v_new
+    assert data.v == v_new
+    assert data.y.v == "test_nested"
+    data.y.v = "test_nested_new"
+    assert data.y.v == "test_nested_new"
+    data_nest.v = "test_nested"
+    assert data_nest.v == "test_nested"
```

Comment:
we miss test setting a tensor of a nested class

Reply:
Added the assertion for setting a tensor

Reply:
don't see `data.y.X=something` anywhere